contentcuration/contentcuration/models.py

```diff
@@ -245,18 +245,21 @@ class Channel(models.Model):
     source_domain = models.CharField(max_length=300, blank=True, null=True)
     ricecooker_version = models.CharField(max_length=100, blank=True, null=True)
 
+    def resource_size_key(self):
+        return "{}_resource_size".format(self.pk)
+
+    # Might be good to display resource size, but need to improve query time first
     def get_resource_size(self):
-        # TODO: Add this back in once query filters out duplicated checksums
-        size = ChannelResourceSize.objects.filter(id=self.main_tree.tree_id).first()
-        if size:
-            return size.resource_size
-        return 0
-
-        # descendants = self.main_tree.get_descendants().prefetch_related('assessment_items')
-        # size_q = File.objects.select_related('contentnode').select_related('assessment_item')\
-        #         .filter(Q(contentnode_id__in=descendants.values_list('id', flat=True)) | Q(assessment_item_id__in=descendants.values_list('assessment_items__id', flat=True)))\
-        #         .values('checksum', 'file_size').distinct().aggregate(resource_size=Sum('file_size'))
-        # return size_q['resource_size'] or 0
+        cached_data = cache.get(self.resource_size_key())
+        if cached_data:
+            return cached_data
+        tree_id = self.main_tree.tree_id
+        files = File.objects.select_related('contentnode').select_related('assessment_item')\
+            .filter(Q(contentnode__tree_id=tree_id) | Q(assessment_item__contentnode__tree_id=tree_id))\
+            .values('checksum', 'file_size').distinct().aggregate(resource_size=Sum('file_size'))
```

Comment:
It might read more easily if each part of this chain were on a newline, so it almost looks like how one might right the actual SQL

```
.values(...) \
.distinct() \
.aggregate()
```